chopper/lib/src/interceptors/request_converter_interceptor.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:chopper/src/annotations.dart';
+import 'package:chopper/src/chain/chain.dart';
+import 'package:chopper/src/converters.dart';
+import 'package:chopper/src/interceptors/internal_interceptor.dart';
+import 'package:chopper/src/request.dart';
+import 'package:chopper/src/response.dart';
+
+/// {@template RequestConverterInterceptor}
+/// Internal interceptor that handles request conversion provided by [_requestConverter] or [_converter].
+/// {@endtemplate}
+class RequestConverterInterceptor implements InternalInterceptor {
+  /// {@macro RequestConverterInterceptor}
+  RequestConverterInterceptor(this._converter, this._requestConverter);
+
+  /// Converter to be used for request conversion.
+  final Converter? _converter;
+
+  /// Request converter to be used for request conversion.
+  final ConvertRequest? _requestConverter;
+
+  @override
+  Future<Response<BodyType>> intercept<BodyType>(Chain<BodyType> chain) async {
+    final request =
+        await _handleRequestConverter(chain.request, _requestConverter);
+
+    final response = await chain.proceed(request);
```

Comment:
Maybe I'm being nitpicky, but this could be shortened to just
```dart
return await chain.proceed(
  await _handleRequestConverter(
    chain.request, 
    _requestConverter,
  ),
);
```